ci/jenkins/test.sh

```diff
@@ -853,24 +853,26 @@ function redeploy_k8s_if_ip_mode_changes() {
     HAS_IPV6=${INITIAL_VALUE}
     POD_CIDRS=($( (kubectl get node ${CONTROL_PLANE_HOSTNAME} -o json | jq -r '.spec.podCIDRs | @sh') | tr -d \'\")) || true
     echo "POD_CIDRS=${POD_CIDRS[*]}"
-    for POD_CIDR in "${POD_CIDRS[@]}"; do
-        if [[ $POD_CIDR =~ .*:.* ]]
-        then
-            (( HAS_IPV6++ ))
+    if [[ -n $POD_CIDRS ]]; then
+        for POD_CIDR in "${POD_CIDRS[@]}"; do
+            if [[ $POD_CIDR =~ .*:.* ]]
+            then
+                (( HAS_IPV6++ ))
+            else
+                (( HAS_IPV4++ ))
+            fi
+        done
+        if [[ ${IP_MODE} == "ipv4" ]]; then
+            (( HAS_IPV4-- ))
+        elif [[ ${IP_MODE} == "ipv6" ]]; then
+            (( HAS_IPV6-- ))
         else
-            (( HAS_IPV4++ ))
+            (( HAS_IPV4-- ))
+            (( HAS_IPV6-- ))
+        fi
+        if [ ${HAS_IPV4} -eq ${INITIAL_VALUE} ] && [ ${HAS_IPV6} -eq ${INITIAL_VALUE} ]; then
+          return 0
```

Comment:
The change is fine but I don't see what's wrong with the previous code too.

Reply:
We added a check `if [[ -n $POD_CIDRS ]]; then` for this code, to skip this check when kubectl cannot get the POD_CIDRS.
Previously the redeployment will be skipped here if k8s is not deployed and POD_CIDRS is empty.